src/components/RevokeContainer/RevokeContainer.tsx

```diff
@@ -0,0 +1,125 @@
+
+      const validateDocument = async () => {
+        const network = config?.network || "homestead";
+        if (network !== "local") {
+          const verify = verificationBuilder(openAttestationVerifiers, { network: network });
+          const fragments = await verify(revokeDocuments[0]);
+          const notRevokeFragments = getNotRevokeFragment(fragments);
+          const hashValid = isValid(fragments, ["DOCUMENT_INTEGRITY"]);
+          const issuedValid = isValid(notRevokeFragments, ["DOCUMENT_STATUS"]);
+          const identityValid = isValid(fragments, ["ISSUER_IDENTITY"]);
+          const isDocumentValid = hashValid && issuedValid && identityValid;
+
+          if (!isDocumentValid) {
+            setErrorMessage("Error: Document is not valid, please use a valid document.");
+            setRevokeDocuments([]);
+            setDocumentUploadState(DocumentUploadState.ERROR);
+          } else {
+            setErrorMessage("");
+            setDocumentUploadState(DocumentUploadState.DONE);
+          }
+        } else {
+          setErrorMessage("");
+          setDocumentUploadState(DocumentUploadState.DONE);
+        }
+      };
+
+      if (!isDocumentRevokable) {
+        setErrorMessage("Error: Document is not revokable, please use a revokable document.");
+        setRevokeDocuments([]);
+        setDocumentUploadState(DocumentUploadState.ERROR);
+      } else {
+        validateDocument();
+      }
+    }
+  }, [config?.network, revokeDocuments]);
+
+  if (!config) {
+    return <Redirect to="/" />;
+  }
+
+  const revokingDocument = () => {
+    setRevokeStep(2);
+    setShowConfirmationModal(false);
+  };
+  const revokeAnotherDocument = () => {
+    setRevokeStep(1);
+    setRevokeDocuments([]);
+    setFileName("");
+    setErrorMessage("");
+    setShowConfirmationModal(false);
+  };
+
+  const onLogout = () => {
+    revokeAnotherDocument();
+    setConfig(undefined);
+  };
+
+  return (
+    <>
+      <RevokeConfirmationModal
+        revokingDocument={revokingDocument}
+        show={showConfirmationModal}
+        closeRevokeConfirmationModal={() => setShowConfirmationModal(false)}
+      />
+      {revokeStep === 1 && (
+        <>
+          {revokeDocuments && revokeDocuments.length > 0 && documentUploadState === DocumentUploadState.DONE ? (
+            <RevokeDocumentTileArea
+              revokeDocuments={revokeDocuments}
+              fileName={fileName}
+              onShowConfirmation={() => setShowConfirmationModal(true)}
+              documentUploadState={documentUploadState}
+              onBack={revokeAnotherDocument}
+            />
+          ) : (
+            <RevokeDocumentDropZone
+              setRevokeDocuments={setRevokeDocuments}
+              errorMessage={errorMessage}
+              setFileName={setFileName}
+              documentUploadState={documentUploadState}
+              setDocumentUploadState={setDocumentUploadState}
+            />
+          )}
```

Comment:
should these 2 be broken into each step? so total 3 steps, 1 = upload document, 2 = confirm document, 3 = revoked document

Reply:
ok can will update it accordingly.